nextest-runner/src/runner/executor.rs

```diff
@@ -573,6 +574,269 @@ impl<'a> ExecutorContext<'a> {
         })
     }
 
+    /// Run a pre-timeoutx script in its own process.
+    #[instrument(level = "debug", skip(self, resp_tx, req_rx))]
+    async fn run_pre_timeout_script<'b>(
+        &self,
+        script: PreTimeoutScriptPacket<'a, 'b>,
+        resp_tx: &UnboundedSender<ExecutorEvent<'a>>,
+        req_rx: &mut UnboundedReceiver<RunUnitRequest<'a>>,
+    ) {
+        let test_instance = script.test.test_instance;
+        let _ = resp_tx.send(ExecutorEvent::PreTimeoutScriptStarted { test_instance });
+
+        let mut stopwatch = crate::time::stopwatch();
+
+        let status = match self
+            .run_pre_timeout_script_inner(script.clone(), &mut stopwatch, resp_tx, req_rx)
+            .await
+        {
+            Ok(status) => status,
+            Err(error) => InternalPreTimeoutScriptExecuteStatus {
+                slow_after: None,
+                output: ChildExecutionOutput::StartError(error),
+                result: ExecutionResult::ExecFail,
+                stopwatch_end: stopwatch.snapshot(),
+            },
+        };
+
+        let _ = resp_tx.send(ExecutorEvent::PreTimeoutScriptFinished {
+            test_instance,
+            script_id: script.script_id,
+            config: script.config,
+            status: status.into_external(),
+        });
+    }
+
+    #[instrument(level = "debug", skip(self, resp_tx, req_rx))]
+    async fn run_pre_timeout_script_inner<'b>(
+        &self,
+        script: PreTimeoutScriptPacket<'a, 'b>,
+        stopwatch: &mut StopwatchStart,
+        resp_tx: &UnboundedSender<ExecutorEvent<'a>>,
+        req_rx: &mut UnboundedReceiver<RunUnitRequest<'a>>,
+    ) -> Result<InternalPreTimeoutScriptExecuteStatus, ChildStartError> {
+        // XXX: this function is 95% the same as run_setup_script_inner. Do we
+        // need to try to factor out a shared helper method?
```

Comment:
@sunshowers would love your quick take on this question.

Reply:
95% the same -- what are the differences? Looking at the function itself, yes -- I'd love to see some common code factored out.